apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoader.java

```diff
@@ -437,6 +442,15 @@ private void processAsync(
           Object recordInDb = entityManager.find(record.getClass(), recordId);
           timerIdempotencyQuery.close();
 
+          /* Blow up the data load if we try to insert a record that has a non 2022 year.
+           * See {@link LoadAppOptions.isFilteringNonNullAndNon2022Benes}
+           */
+          if (options.isFilteringNonNullAndNon2022Benes()
+              && !isBeneficiaryWithNullOr2022Year(rifRecordEvent)) {
+            throw new IllegalArgumentException(
+                "Cannot INSERT beneficiary with non-2022 enrollment year; investigate this data load.");
```

Comment:
Might also want a log message here; Im pretty sure this exception gets swallowed upstream and doesnt get shown in the log like I wanted.

Reply:
I would have expected that it would end up in the catch block and log the warning on line 537. no? Does the pipeline actually abort on this condition?

Reply:
It's working in the ITs, at least:

```
java.lang.IllegalArgumentException: Cannot INSERT beneficiary with non-2022 enrollment year; investigate this data load.
	at gov.cms.bfd.pipeline.ccw.rif.load.RifLoader.process(RifLoader.java:471)
	at gov.cms.bfd.pipeline.ccw.rif.load.RifLoader.lambda$2(RifLoader.java:353)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
2022-03-04 13:44:32,923 [pool-2-thread-1] WARN  g.c.b.p.ccw.rif.load.RifLoaderIT - Record(s) failed to load.
gov.cms.bfd.pipeline.ccw.rif.load.RifLoadFailure: Failed to load 'BENEFICIARY' records.
	at gov.cms.bfd.pipeline.ccw.rif.load.RifLoader.process(RifLoader.java:539)
	at gov.cms.bfd.pipeline.ccw.rif.load.RifLoader.lambda$2(RifLoader.java:353)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalArgumentException: Cannot INSERT beneficiary with non-2022 enrollment year; investigate this data load.
	at gov.cms.bfd.pipeline.ccw.rif.load.RifLoader.process(RifLoader.java:471)
	... 6 common frames omitted
```

And if exceptions are being swallowed in the combined/assembled app, we should file that as a bug, rather than working around it.